### PR TITLE
[python][c++] Correctly manage lifetime of `SOMAVFS` with `SOMAVFSFilebuf`

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue and/or context:**

[[sc-65831](https://app.shortcut.com/tiledb-inc/story/65831/python-c-segv-memory-issues-in-somavfs-somavfsfilebuf)]

**Changes:**

Modify `SOMAVFSFilebuf` to hold a smart pointer to `SOMAVFS`.

### CODE REVIEW

File: apis/python/tests/test_basic_anndata_io.py [hunk 1]
```diff
@@ -1557,3 +1557,16 @@ def test_from_anndata_byteorder_63459(tmp_path, conftest_pbmc_small):
...
+
+def test_vfs_lifetime_65831():
+    # https://app.shortcut.com/tiledb-inc/story/65831/python-c-segv-memory-issues-in-somavfs-somavfsfilebuf
+    context = tiledbsoma.SOMATileDBContext()
+    vfs = tiledbsoma.pytiledbsoma.SOMAVFS(context.native_context)
+    fb = tiledbsoma.pytiledbsoma.SOMAVFSFilebuf(vfs).open(
+        str(TESTDATA / "pbmc-small.h5ad")
+    )
+    del vfs
+    # Ensure that read does not segfault
```
Comment (johnkerl) [inline apis/python/tests/test_basic_anndata_io.py:1570]:
```suggestion
    # Implicitly ensure that read does not segfault
```

**Review by johnkerl — APPROVED** [review]
🚢 

Thanks @nguyenv !

File: apis/python/tests/test_basic_anndata_io.py [hunk 2]
```diff
@@ -1557,3 +1557,16 @@ def test_from_anndata_byteorder_63459(tmp_path, conftest_pbmc_small):
...
+
+
+def test_vfs_lifetime_65831():
+    # https://app.shortcut.com/tiledb-inc/story/65831/python-c-segv-memory-issues-in-somavfs-somavfsfilebuf
+    context = tiledbsoma.SOMATileDBContext()
+    vfs = tiledbsoma.pytiledbsoma.SOMAVFS(context.native_context)
+    fb = tiledbsoma.pytiledbsoma.SOMAVFSFilebuf(vfs).open(
+        str(TESTDATA / "pbmc-small.h5ad")
+    )
+    del vfs
```
Comment (bkmartinjr) [inline apis/python/tests/test_basic_anndata_io.py:1569]: Optional: you might consider explicitly calling `gc.collect()` here, just to cover the edge case where the garbage collector doesn't immediately free `vfs`

**Review by bkmartinjr — APPROVED** [review]
